src/tlo/analysis/life_expectancy.py

```diff
@@ -0,0 +1,305 @@
+            output_from_eval: pd.Series = df.loc[pd.to_datetime(df.date).between(*TARGET_PERIOD)].stack()
+            res.loc[:, (draw, run)] = output_from_eval[1:]
+
+    return res
+
+
+def _get_population_size_by_age(results_folder):
+    def extract_infants_during_target_period(df: pd.DataFrame) -> pd.Series:
+        return df.loc[pd.to_datetime(df.date).between(*TARGET_PERIOD), '0']
+
+    # get children aged=0 in target period
+    num_children = extract_results(
+        results_folder,
+        module="tlo.methods.demography",
+        key="num_children",
+        custom_generate_series=extract_infants_during_target_period,
+        do_scaling=False
+    )
+    num_f0 = num_children / 2  # todo should this be exactly 50/50?
+    num_m0 = num_children / 2
+
+    # get age_range_m and age_range_f for all age-groups
+    num_males = _extract_pop_during_target_period(results_folder=results_folder, key='age_range_m')
+    num_females = _extract_pop_during_target_period(results_folder=results_folder, key='age_range_f')
+    num_males = num_males.reset_index(level=0, drop=True)
+    num_females = num_females.reset_index(level=0, drop=True)
+
+    # then subtract age 0 from age-group 0-4
+    num_males_adj = num_males.iloc[0:1, :] - num_m0.iloc[0:1, :].values
+    # add adjusted data back into dataframe
+    num_males.iloc[0] = num_males_adj.values
+    # change first row index
+    num_males.index.values[0] = '1-4'
+    # add in infant numbers to main dataframe
+    t1 = pd.concat([num_m0, num_males])
+    # set correct index
+    t1.index.values[0] = '0'
+
+    # repeat for females
+    num_females_adj = num_females.iloc[0:1, :] - num_f0.iloc[0:1, :].values
+    num_females.iloc[0] = num_females_adj.values
+    num_females.index.values[0] = '1-4'
+    t2 = pd.concat([num_f0, num_females])
+    t2.index.values[0] = '0'
+
+    # combine pop aged 90+
+    # Sum the last three rows column-wise
+    last_three_rows_sum = t1.iloc[-3:].sum()
+    t1.loc['90+'] = last_three_rows_sum
+
+    last_three_rows_sum = t2.iloc[-3:].sum()
+    t2.loc['90+'] = last_three_rows_sum
+
+    # remove unneeded columns
+    indices_to_remove = ['90-94', '95-99', '100+']
+    t1 = t1.drop(indices_to_remove, axis=0)
+    t2 = t2.drop(indices_to_remove, axis=0)
+
+    # output is 0, 1-4, 5-9, 10-14 and index by male/female
+    # join male and female dataframe and add index for sex
+    t3 = pd.concat([t1, t2], ignore_index=True)
+
+    # Create a MultiIndex for rows using age group and 'Male' and 'Female'
+    multi_index_rows_male = pd.MultiIndex.from_product([t1.index, ['M']], names=['age_group', 'sex'])
+    multi_index_rows_female = pd.MultiIndex.from_product([t1.index, ['F']], names=['age_group', 'sex'])
+    t3.index = multi_index_rows_male.append(multi_index_rows_female)
+
+    return t3
+
+
+# %% GENERATE LIFE EXPECTANCY ESTIMATES
+
+def _estimate_life_expectancy(person_years_at_risk, number_of_deaths_in_interval):
```

Comment:
naming of this argument is a bit confusing, as it's actually population size (although could be PY from the demography module)